testing/internal/e2e/docker/docker.go

```diff
@@ -0,0 +1,236 @@
+package docker
+
+import (
+	"context"
+	"embed"
+	"fmt"
+	"io"
+	"os"
+	"path"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"github.com/testcontainers/testcontainers-go"
+	"github.com/testcontainers/testcontainers-go/wait"
+	"golang.org/x/crypto/ssh"
+)
+
+//go:embed testdata/boundary-config.hcl
+var ControllerConfigFile embed.FS
+
+//go:embed testdata/postgresql.conf
+var postgresConfigFile embed.FS
+
+type e2eContainer struct {
+	testcontainers.Container
+	UriNetwork   string
+	UriLocalhost string
+}
+
+// CreateBoundaryDatabase spins up a postgres database in a docker container
+// Returns information about the container
+func CreateBoundaryDatabase(t testing.TB, networkName string) *e2eContainer {
+	t.Log("Starting postgres database...")
+
+	data, err := postgresConfigFile.ReadFile("testdata/postgresql.conf")
+	require.NoError(t, err)
+	postgresConfigFilePath := fmt.Sprintf("%s/%s", t.TempDir(), "postgresql.conf")
+	f, err := os.Create(postgresConfigFilePath)
+	require.NoError(t, err)
+	_, err = f.WriteString(string(data))
+	require.NoError(t, err)
```

Comment:
why do you need this temp file? Can you not just mount this file from `testdata/postgresql.conf` into the container?